packages/hoprd/crates/hoprd-misc/src/cli.rs

```diff
@@ -0,0 +1,457 @@
+        .arg(Arg::new("identity")
+            .long("identity")
+            .help("The path to the identity file")
+            .env("HOPRD_IDENTITY")
+            .default_value(format!("{}/.hopr-identity", home_path)))
+        .arg(Arg::new("data")
+            .long("data")
+            .help("manually specify the data directory to use")
+            .env("HOPRD_DATA")
+            .default_value(get_data_path(mono_repo_path, maybe_default_environment)));
+
+        // Add compile args to runtime-time args
+        cmd = Self::augment_args(cmd);
+
+        cmd.update_env_from(env_vars);
+
+        let derived_matches = cmd
+            .try_get_matches_from(cli_args)
+            .map_err(|e| e.to_string())?;
+
+        let mut args = ok_or_str!(Self::from_arg_matches(&derived_matches))?;
+
+        args.augment_runtime_args(&derived_matches);
+
+        Ok(args)
+    }
+}
+
+#[derive(Deserialize)]
+struct DefaultEnvironmentFile {
+    id: String,
+}
+
+impl FromJsonFile for DefaultEnvironmentFile {
+    fn from_json_file(mono_repo_path: &str) -> Result<Self, String> {
+        let default_environment_json_path: String =
+            format!("{}/default-environment.json", mono_repo_path);
+        let data = ok_or_str!(real::read_file(default_environment_json_path.as_str()))?;
+
+        ok_or_str!(serde_json::from_slice::<DefaultEnvironmentFile>(&data))
+    }
+}
+
+/// Checks for `default_environment.json` file and, if present, returns their environment id
+fn get_default_environment(mono_repo_path: &str) -> Option<String> {
+    match DefaultEnvironmentFile::from_json_file(mono_repo_path) {
+        Ok(json) => Some(json.id),
+        Err(_) => None,
+    }
+}
+
+/// Gets the default path where the database is stored at
+fn get_data_path(mono_repo_path: &str, maybe_default_environment: Option<String>) -> String {
+    match maybe_default_environment {
+        Some(default_environment) => format!(
+            "{}/packages/hoprd/hoprd-db/{}",
+            mono_repo_path, default_environment
+        ),
+        None => format!("{}/packages/hoprd/hoprd-db", mono_repo_path),
+    }
+}
+
+pub mod wasm {
+    use js_sys::JsString;
+    use std::collections::HashMap;
+    use std::ffi::OsString;
+    use std::str::FromStr;
+    use wasm_bindgen::prelude::*;
+    use wasm_bindgen::JsValue;
+
+    /// Macro used to convert Vec<JsString> to Vec<&str>
+    macro_rules! convert_from_jstrvec {
+        ($v:expr,$r:ident) => {
+            let _aux: Vec<String> = $v.iter().map(String::from).collect();
+            let $r = _aux.iter().map(AsRef::as_ref).collect::<Vec<&str>>();
+        };
+    }
+
+    macro_rules! clean_mono_repo_path {
+        ($v:expr,$r:ident) => {
+            let $r = $v.strip_suffix("/").unwrap_or($v);
+        };
+    }
+
+    macro_rules! ok_or_jserr {
+        ($v:expr) => {
+            $v.map_err(|e| JsValue::from(e.to_string()))
+        };
+    }
+
+    #[wasm_bindgen]
+    pub fn parse_cli_arguments(
```

Comment:
Since this function contains some non-trivial logic, let's maybe put a simple unit test for it into `tests/node.rs`.

Reply:
You're right. This is worth a unit test. Haven't done this because this function assumes the existence of quite a bunch of configuration files, that I don't want to fake for a unit test